validphys2/src/validphys/plotutils.py

```diff
@@ -13,45 +13,107 @@
 import scipy.stats as stats
 
 import numpy as np
-import matplotlib.pyplot as plt
+from matplotlib.figure import Figure
+import matplotlib as mpl
 import matplotlib.scale as mscale
 import matplotlib.patches as mpatches
 import matplotlib.collections as mcollections
-from matplotlib  import transforms
+from matplotlib import transforms
 from matplotlib.markers import MarkerStyle
 from matplotlib import ticker
 
 from reportengine.floatformatting import format_number
 
 log = logging.getLogger(__name__)
 
+DEFAULT_FIGSIZE = (6.4, 4.8)
+
+
+def subplots(figsize=(6.4, 4.8), nrows=1, ncols=1, sharex=False, sharey=False, **kwargs):
+    """
+    matplotlib.figure wrapper used to generate subplots.
+
+    Use matplotlib.figure.Figure() objects to avoid importing ``pyplot`` anywhere.
+    The reason is that ``pyplot`` maintains a global state that makes it misbehave
+    in multithreaded applications such when executed under dask parallel mode.
+
+    Parameters
+    ----------
+    figsize : 2-tuple of floats
+            defaults is (6.4, 4.8), that is, the same that would be set by mpl.rcParams["figure.figsize"]
+            avoids global figsize state set by rcParams
+    nrows, ncols : int, default 1
+
+    sharex, sharey : bool, default False
+
+    Returns
+    -------
+    tuple
+        fig, ax = (matplotlib.figure.Figure, fig.subplots)
+    """
+    fig = Figure(figsize=figsize)
+    ax = fig.subplots(nrows=nrows, ncols=ncols, sharex=sharex, sharey=sharey, **kwargs)
+    return fig, ax
+
+
+def add_subplot(figsize=(6.4, 4.8), projection=None, **kwargs):
+    """
+    matplotlib.figure wrapper used to add subplots.
+
+    Use matplotlib.figure.Figure() objects to avoid importing ``pyplot`` anywhere.
+    The reason is that ``pyplot`` maintains a global state that makes it misbehave
+    in multithreaded applications such when executed under dask parallel mode.
+
+    Parameters
+    ----------
+    figsize : 2-tuple of floats
+            default is (6.4, 4.8), that is, the same that would be set by mpl.rcParams["figure.figsize"]
+            avoids global figsize state set by rcParams
+    projections : The projection type of the subplot (Axes).
+                default is None
+
+    Returns
+    -------
+    tuple
+        fig, ax = (matplotlib.figure.Figure, fig.add_subplot)
+    """
+    fig = Figure(figsize=figsize)
+    ax = fig.add_subplot(projection=projection, **kwargs)
+    return fig, ax
```

Comment:
I think it might be worth to revert the changes back to this commit: https://github.com/NNPDF/nnpdf/pull/1723/commits/8f78b4891ed46fb483beb0f326d240c121558a3b

And then, for both functions, pass the arguments through subplots as `**kwargs`. This should leave the default as given in the style file.

I would still like `black` to be run for these two functions but I'd avoid running it for files where the only change was `plt.subplots` -> `plotutils.subplots`

(as a separate note, I didn't realize we had files that were mixing 8 and 4 spaces for the indentation... I'm still convinced we should do a commit with a run of black for the entire code, but it should not be this PR)

Reply:
Yes. I know (and I can volunteer to eventually do it). But there are some other (like mixing 4/8 spaces) that I'd like to fix for my own sanity like this function [here](https://github.com/NNPDF/nnpdf/blob/f02b49a8e6eb785af6a56dc3195133abff571046/validphys2/src/validphys/plotutils.py#L51)

Reply:
But if you know there are specific points in which you don't want Black to act, why don't you just mark them explicitly [`# fmt: on/off`](https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html)?

Reply:
The problem is the same, one person has to go through the codebase to do so. There are files I've never opened there!

Reply:
True, but it can also be faced incrementally: if you already know that there are parts of `pdfbases.py` that should not be touched by Black, you could start marking them.

This would be useful even for other people working on it (e.g. I have Black running on save, even though if I notice that it is massively reformatting a file beyond my changes I manually skip it)